designate/central/service.py

```diff
@@ -3509,3 +3511,89 @@ def delete_service_status(self, context, service_status):
                     'hostname': service_status.hostname
                 }
             )
+
+    # Shared pools
+    @rpc.expected_exceptions()
+    @notification.notify_type('dns.pool.share')
+    @transaction
+    def share_pool(self, context, pool_id, shared_pool):
+        # Ensure that pool exists and get the pool owner
+        elevated_context = context.elevated(all_tenants=True)
+        pool = self.storage.get_pool(elevated_context, pool_id)
+        domain_id = pool.domain_id
+        if not domain_id:
+            raise exceptions.PoolWithoutDomainId(
+                f"Pools {pool_id} doesn't have domain_id"
+                f"Specify domain_id in pools.yaml.")
+        if policy.enforce_new_defaults():
+            target = {constants.RBAC_DOMAIN_ID: domain_id}
+        else:
+            target = {'domain_id': domain_id}
+
+        policy.check('share_pool', context, target)
+        shared_pool['domain_id'] = pool.domain_id
+        shared_pool['pool_id'] = pool_id
+
+        shared_pool = self.storage.share_pool(context, shared_pool)
+
+        return shared_pool
+
+    @rpc.expected_exceptions()
+    @notification.notify_type('dns.pool.unshare')
+    @transaction
+    def unshare_pool(self, context, pool_id, pool_share_id):
+        # Ensure the share exists and get the share owner
+        shared_pool = self.get_shared_pool(context, pool_id, pool_share_id)
+
+        if policy.enforce_new_defaults():
+            target = {constants.RBAC_PROJECT_ID: shared_pool.domain_id}
+        else:
+            target = {'domain_id': shared_pool.domain_id}
+
+        policy.check('unshare_pool', context, target)
```

Comment:
Does this actually work for a non-admin user? Because target is always `project_id`, but here it is `domain_id`.